### PR TITLE
Add a11y tests for required attrs and attr-value

### PR DESCRIPTION
**Pull request by robdodson**
- Adds the above two new audits
- Tidy existing a11y tests to make them consistent
- Add a11y test page

@ebidel PTAL

### CODE REVIEW

**Comment by paulirish** [comment]
Just tested and this PR and #1155 appear to be in harmony.

@robdodson can you give me an lgtm on 1155 if it's looking on point?

File: lighthouse-cli/test/fixtures/a11y/a11y_tester.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
```
Comment (brendankenny) [inline lighthouse-cli/test/fixtures/a11y/a11y_tester.html]: needs a license header

Comment (ebidel) [inline lighthouse-cli/test/fixtures/a11y/a11y_tester.html]: Can you add the license header? Copy from the DBW tester.

File: lighthouse-core/test/audits/aria-required-attr-test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../../audits/aria-required-attr.js');
+const assert = require('assert');
+
+/* global describe, it*/
```
Comment (brendankenny) [inline lighthouse-core/test/audits/aria-required-attr-test.js]: can use `/* eslint-env mocha */` instead

File: lighthouse-core/audits/aria-required-attr.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const Formatter = require('../formatters/formatter');
+
```
Comment (ebidel) [inline lighthouse-core/audits/aria-required-attr.js]: can you add a description to the top of this file to explain what the audit is? Look at the dbw audits for examples.

File: lighthouse-core/audits/aria-required-attr.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+class ARIARequiredAttr extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Accessibility',
+      name: 'aria-required-attr',
+      description: 'Elements with ARIA roles have all required aria-* attributes',
```
Comment (ebidel) [inline lighthouse-core/audits/aria-required-attr.js]: all -> the

File: lighthouse-core/audits/aria-valid-attr-value.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const Formatter = require('../formatters/formatter');
+
```
Comment (ebidel) [inline lighthouse-core/audits/aria-valid-attr-value.js]: add description.

File: lighthouse-core/audits/aria-valid-attr-value.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+class ARIAValidAttr extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Accessibility',
+      name: 'aria-valid-attr-value',
+      description: 'Element aria-* attributes have valid values',
```
Comment (ebidel) [inline lighthouse-core/audits/aria-valid-attr-value.js]: "Elements with aria-* attributes have valid values"? Maybe elaborate a bit more. A noob like me doesn't know what "valid values" mean.

File: lighthouse-core/test/audits/aria-valid-attr-value-test.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../../audits/aria-valid-attr-value.js');
+const assert = require('assert');
+
+/* global describe, it*/
```
Comment (brendankenny) [inline lighthouse-core/test/audits/aria-valid-attr-value-test.js]: `/* eslint-env mocha */`

File: lighthouse-core/audits/aria-valid-attr-value.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      rawValue: typeof rule === 'undefined',
+      debugString: this.createDebugString(rule),
+      extendedInfo: {
+        formatter: Formatter.SUPPORTED_FORMATS.ACCESSIBILITY,
+        value: rule
+      }
+    });
+  }
+
+  static createDebugString(rule) {
```
Comment (ebidel) [inline lighthouse-core/audits/aria-valid-attr-value.js]: This `createDebugString` method begs for a refactor helper :) ...and a lot of these audits.

But for starters, you could factor our `createDebugString` into lib/a11y-helpers.js like I did for common style audit methods:
https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/lib/styles-helpers.js

File: lighthouse-cli/test/fixtures/a11y/a11y_tester.html (hunk 1 above)
Comment (brendankenny) [inline lighthouse-cli/test/fixtures/a11y/a11y_tester.html]: is this just for local testing right now? Is the plan to make something like the dbw smoke tests?

File: lighthouse-core/test/audits/aria-required-attr-test.js (hunk 2 above)
Comment (ebidel) [inline lighthouse-core/test/audits/aria-required-attr-test.js]: `/* eslint-env mocha */`

File: lighthouse-core/test/audits/aria-valid-attr-value-test.js (hunk 7 above)
Comment (ebidel) [inline lighthouse-core/test/audits/aria-valid-attr-value-test.js]: `/* eslint-env mocha */`

**Comment by brendankenny** [comment]
axe 2.1.7 ready for you @robdodson

File: lighthouse-cli/test/fixtures/a11y/a11y_tester.html (hunk 1 above)
Comment (robdodson) [inline lighthouse-cli/test/fixtures/a11y/a11y_tester.html]: local for now. will add smokehouse in a future iteration

**Comment by robdodson** [comment]
@ebidel @brendankenny PTAL

File: lighthouse-core/audits/aria-allowed-attr.js [hunk 9]
```diff
@@ -17,7 +17,12 @@
 
 'use strict';
 
+/**
+ * @fileoverview Ensures ARIA attributes are allowed for an element's role
```
Comment (ebidel) [inline lighthouse-core/audits/aria-allowed-attr.js]: Can you end all these descriptions with a period? I hate to be that guy, but it like not having a close brace in code!

File: lighthouse-core/lib/a11y-helpers.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * Creates an output string to tell the user why their audit failed
```
Comment (ebidel) [inline lighthouse-core/lib/a11y-helpers.js]: failed.

File: lighthouse-core/lib/a11y-helpers.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * @param {{nodes: !Array, help: string}} rule An aXe test rule with a list of failing nodes
+ *    and help text on how to fix the violation
```
Comment (ebidel) [inline lighthouse-core/lib/a11y-helpers.js]: violation.

File: lighthouse-core/lib/a11y-helpers.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+ * @return {string} A debug string telling the user why their test failed and which nodes
+ *    caused the failure
```
Comment (ebidel) [inline lighthouse-core/lib/a11y-helpers.js]: failure.

File: lighthouse-core/audits/aria-allowed-attr.js (hunk 9 above)
Comment (robdodson) [inline lighthouse-core/audits/aria-allowed-attr.js]: this guy....

**Comment by robdodson** [comment]
periods are in!

**Pull request merged**